andela/boondocks-bn-backend
# #169257397 User can login via email and password

#### What does this PR do?
This PR creates a login functionality for users to get access to the application by supplying their email and password

#### Description of Task to be completed?
- [x] Setup/refactor user model, route and controller
- [x] Setup jwtToken and Bcrypt
- [x] Setup seeder file to populate test user record in the database
- [x] Successfully link route with validation package

#### How should this be manually tested?
Option 1
- Clone this repo and navigate to the project folder using Terminal
- Install dependencies using the command `npm install`
- Run the command `npm test` on Terminal

Option 2
- Run the command `npx sequelize-cli db:seed:all` to add a test user to your database
- Using Postman, create a POST request to the URL: (http://localhost:3000/api/v1/auth/signin) with the following credentials { email: 'john@barefoot.com',  password: '12345678' }
- You should receive a success response, change the credentials to get an error message

#### Any background context you want to provide?
N/A

#### What are the relevant pivotal tracker stories?
[#169257397](https://www.pivotaltracker.com/story/show/169257397)

#### Screenshots (if appropriate)
<img width="1435" alt="successfulLogin" src="https://user-images.githubusercontent.com/17108099/68809158-f9ca1e80-0673-11ea-84ad-3cef012d85f4.png">


#### Questions:
N/A
